openmole/plugins/org.openmole.plugin.tool.pattern/src/main/scala/org/openmole/plugin/tool/pattern/Strain.scala

```diff
@@ -28,7 +29,14 @@ object Strain {
     val first = Capsule(EmptyTask(), strainer = true)
     val last = Slot(Capsule(EmptyTask(), strainer = true))
 
-    (first -- puzzle -- last) + (first -- last)
+    val mole = puzzle.toMole
+    val sources = Sources(puzzle.sources.groupBy(_._1).mapValues(_.map(_._2)))
+    val hooks = Hooks(puzzle.hooks.groupBy(_._1).mapValues(_.map(_._2)))
```

Comment:
From Puzzle.scala:
```scala
case class Puzzle(
    firstSlot: Slot,
    lasts: Iterable[Capsule] = Iterable.empty,
    transitions: Iterable[ITransition] = Iterable.empty,
    dataChannels: Iterable[DataChannel] = Iterable.empty,
    sources: Iterable[(Capsule, Source)] = Iterable.empty,
    hooks: Iterable[(Capsule, Hook)] = Iterable.empty,
    environments: Map[Capsule, Environment] = Map.empty,
    grouping: Map[Capsule, Grouping] = Map.empty)
```
whereas the constructor of ```Sources```:
```scala
case class Sources(map: Map[Capsule, Traversable[Source]])
```
While the ```puzzle.sources``` can contain multiple elements with first tuple entry being the same capsule, this isn't the case for the map. Therefore, the ```groupBy```. Then I just get rid of the thereby introduced duplicate entries in the tuples.